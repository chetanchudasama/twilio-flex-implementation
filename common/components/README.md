# Common Components Library

The common components library contains UI elements and models that can be used across multiple Flex plugins.

It is written in react v16.13.1 and typescript v3.

## Contents

## Pre-requisites

See Pre-requisites section in the [main README.md file](../../README.md).

## Building the library

See Scripts section in the [main README.md file](../../README.md) for building and running the unit tests for this library.
When building the library, the components contained within will be compiled to CommonJS and ESM formats and a `lib` folder will be created containing the compiled files. The Flex plugins will then be able to reference any of the components in the library.
