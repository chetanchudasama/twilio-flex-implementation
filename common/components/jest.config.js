module.exports = {
	watchPlugins: [
		"jest-watch-typeahead/filename",
		"jest-watch-typeahead/testname",
	],
	verbose: true,
	moduleFileExtensions: ["js", "jsx", "ts", "tsx"],
	moduleDirectories: ["node_modules", "src"],
	moduleNameMapper: {
		"\\.(css|less|scss)$": "identity-obj-proxy",
	},
	modulePathIgnorePatterns: ["<rootDir>/lib/"],
	transform: {
		"^.+\\.(ts|tsx)?$": "ts-jest",
	},
	globals: {
		window: true,
		navigator: {
			userAgent: "node",
		},
	},
	coveragePathIgnorePatterns: ["node_modules", ".min.js", "tests", "utils"],
	setupFilesAfterEnv: ["<rootDir>/src/setupTests.ts"],
};
