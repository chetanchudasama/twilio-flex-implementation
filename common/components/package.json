{
  "name": "@common/components",
  "version": "0.1.0",
  "main": "lib/cjs/index.js",
  "typings": "lib/cjs/index.d.ts",
  "module": "lib/esm/index.js",
  "scripts": {
    "test": "jest"
  },
  "dependencies": {
    "@date-io/date-fns": "^1.3.13",
    "@material-ui/core": "^3.9.3",
    "@material-ui/icons": "^3.0.2",
    "@material-ui/lab": "^3.0.0-alpha.30",
    "material-ui-flat-pagination": "^3.2.1",
    "material-ui-pickers": "^2.2.4",
    "moment": "^2.29.1",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-image-lightbox": "^5.1.1",
    "react-emotion": "^9.2.6",
    "date-fns": "^2.22.1"
  },
  "devDependencies": {
    "@types/enzyme": "^3.10.8",
    "@types/enzyme-adapter-react-16": "^1.0.6",
    "@types/jest": "^26.0.23",
    "@types/react": "16.9.56",
    "@types/react-test-renderer": "16.9.5",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "1.15.6",
    "flush-promises": "^1.0.2",
    "jest": "^26.6.3",
    "react-test-renderer": "16.13.1",
    "ts-jest": "^26.5.6"
  },
  "peerDependencies": {
    "react": "16.13.1",
    "react-dom": "16.13.1"
  }
}
