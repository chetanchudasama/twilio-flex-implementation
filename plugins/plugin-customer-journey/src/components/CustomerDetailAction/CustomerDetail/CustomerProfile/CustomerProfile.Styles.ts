import { styled } from "@twilio/flex-ui";

export const CustomerProfileStyles = styled("div")({
  ".card-content": {
    width: "100%",
    margin: 5,
    paddingTop: 0,
  },
  ".card-action": {
    width: "100%",
  },
  ".customer-detail-title": {
    fontSize: "20px",
    fontWeight: 600,
  },
  ".update-detail-btn": {
    float: "right",
  },
  ".birthDate": {
    margin: "8px 0 4px 0",
  },
  ".birthDate .date-picker div": {
    paddingRight: 0,
  },
  ".birthDate .date-picker .clear-icon": {
    marginRight: 3,
  },
  ".birthDate .date-picker input": {
    padding: "15px 14px",
  },
  ".birthDate .picker p": {
    margin: "4px 0 0 0",
  },
});
