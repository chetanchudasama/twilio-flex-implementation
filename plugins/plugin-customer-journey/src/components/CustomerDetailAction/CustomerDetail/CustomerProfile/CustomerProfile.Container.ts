import { connect } from "react-redux";
import { Dispatch } from "redux";
import { CustomerDetailModel } from "@common/components";
import { AppState, namespace } from "../../../../states/index";
import { StateToProps, DispatchToProps } from "./CustomerProfile.Props";
import { CustomerProfile } from "./CustomerProfile";

const mapStateToProps = (state: AppState): StateToProps => ({
  customer: state[namespace].customer.customer ?? new CustomerDetailModel(),
});

const mapDispatchToProps = (dispatch: Dispatch<any>): DispatchToProps => ({});

export default connect(mapStateToProps, mapDispatchToProps)(CustomerProfile);
