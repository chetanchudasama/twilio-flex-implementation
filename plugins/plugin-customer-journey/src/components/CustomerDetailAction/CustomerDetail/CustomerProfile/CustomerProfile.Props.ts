import { CustomerDetailModel } from "@common/components";

export interface StateToProps {
  customer: CustomerDetailModel;
}

export interface DispatchToProps {}
