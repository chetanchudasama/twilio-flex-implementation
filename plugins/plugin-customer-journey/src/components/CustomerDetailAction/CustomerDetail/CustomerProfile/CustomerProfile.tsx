import {
  CustomerDetailModel,
  CustomInput,
  CustomSelect,
  MuiDatePicker,
  Shared,
  TitleModel,
  MaritalStatusModel,
  DrivingLicenceTypeModel,
} from "@common/components";
import React, { ChangeEvent, useState } from "react";
import { Button, Grid, MenuItem } from "@material-ui/core";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";

import { CustomerProfileStyles } from "./CustomerProfile.Styles";
import { StateToProps, DispatchToProps } from "./CustomerProfile.Props";

export interface CustomerProfileProps {
  errorList: Record<string, string[]>;
  updateProfile: (customerDetail: CustomerDetailModel) => void;
}

export type ComponentProps = StateToProps &
  DispatchToProps &
  CustomerProfileProps;

export const CustomerProfile: React.FC<ComponentProps> = ({
  customer,
  errorList,
  updateProfile,
}) => {
  const [customerProfile, setCustomerProfile] =
    useState<CustomerDetailModel>(customer);
  const [errors, setErrors] = useState<Record<string, string[]>>(errorList);
  const [highlightRequiredDetail, setHighlightRequiredDetail] = useState(false);

  const emailRegex = /\S+@\S+\.\S+/;

  const handleCustomerProfileChange = (
    value: string | Date | number | null,
    property: string
  ) => {
    if (highlightRequiredDetail) {
      if (property === "firstName") {
        errors.firstName = value ? [] : ["The first name field is required."];
      }
      if (property === "lastName") {
        errors.lastName = value ? [] : ["The last name field is required."];
      }
      if (property === "maritalStatus") {
        errors.maritalStatus =
          value !== -1 ? [] : ["The marital status field is required."];
      }
      if (property === "drivingLicenceType") {
        errors.drivingLicenceType =
          value !== -1 ? [] : ["The driving license type field is required."];
      }
      if (property === "emailAddress") {
        if (value) {
          errors.emailAddress = !emailRegex.test(value.toString())
            ? ["The email field must be a valid email."]
            : [];
        } else {
          errors.emailAddress = ["The email field is required."];
        }
      }
      setErrors(errors);
    }
    let updatedValue:
      | string
      | Date
      | number
      | null
      | TitleModel
      | MaritalStatusModel
      | DrivingLicenceTypeModel;

    if (property === "title") {
      const selectedTitle = Shared.titles.find(
        (title) => title.id === Number(value)
      );
      updatedValue = {
        titleId: selectedTitle?.id ?? -1,
        titleName: selectedTitle?.description ?? "",
      };
    } else if (property === "maritalStatus") {
      const maritalStatus = Shared.maritalStatuses.find(
        (status) => status.id === Number(value)
      );
      updatedValue = {
        maritalStatusId: maritalStatus?.id ?? -1,
        maritalStatusName: maritalStatus?.description ?? "",
      };
    } else if (property === "drivingLicenceType") {
      const licenceType = Shared.licenceTypes.find(
        (drivingLicenceType) => drivingLicenceType.id === Number(value)
      );
      updatedValue = {
        drivingLicenceTypeId: licenceType?.id ?? -1,
        drivingLicenceTypeName: licenceType?.description ?? "",
      };
    } else {
      updatedValue = value;
    }

    setCustomerProfile({
      ...customerProfile,
      fullName: customerProfile.fullName,
      primaryAddress: customerProfile.primaryAddress,
      address: customerProfile.address,
      [property]: updatedValue,
    });
  };

  return (
    <CustomerProfileStyles>
      <Card>
        <Grid container>
          <CardContent className="card-content">
            <Grid item xs={8} className="customer-detail-title">
              Customer Details
            </Grid>
            <Grid item xs={8}>
              <CustomSelect
                value={customerProfile.title.titleId ?? -1}
                onChange={(event: ChangeEvent<HTMLSelectElement>) =>
                  handleCustomerProfileChange(
                    event.target.value as unknown as number,
                    "title"
                  )
                }
                label="Title"
              >
                <MenuItem value={-1}>
                  <em>Select title</em>
                </MenuItem>
                {Shared.titles.map((title) => {
                  return (
                    <MenuItem key={title.description} value={title.id}>
                      {title.description}
                    </MenuItem>
                  );
                })}
              </CustomSelect>
            </Grid>
            <Grid item xs={8}>
              <CustomInput
                label="First name"
                onChange={(event: ChangeEvent<HTMLInputElement>) =>
                  handleCustomerProfileChange(event.target.value, "firstName")
                }
                value={customerProfile.firstName}
                errors={errors.firstName}
                isRequired
              />
            </Grid>
            <Grid item xs={8}>
              <CustomInput
                label="Middle name(s)"
                onChange={(event: ChangeEvent<HTMLInputElement>) =>
                  handleCustomerProfileChange(event.target.value, "middleName")
                }
                value={customerProfile.middleName}
              />
            </Grid>
            <Grid item xs={8}>
              <CustomInput
                label="Last name"
                onChange={(event: ChangeEvent<HTMLInputElement>) =>
                  handleCustomerProfileChange(event.target.value, "lastName")
                }
                value={customerProfile.lastName}
                errors={errors.lastName}
                isRequired
              />
            </Grid>
            <Grid item xs={8} className="birthDate">
              <MuiDatePicker
                value={customerProfile.dateOfBirth}
                label="D.O.B"
                handleDateChange={(value: Date | null) =>
                  handleCustomerProfileChange(value, "dateOfBirth")
                }
                disableFutureDates
                isRequired
                error={!customerProfile.dateOfBirth && highlightRequiredDetail}
                isOutlinedVariant
              />
            </Grid>
            <Grid item xs={8}>
              <CustomSelect
                value={customerProfile.maritalStatus?.maritalStatusId ?? -1}
                onChange={(event: ChangeEvent<HTMLSelectElement>) =>
                  handleCustomerProfileChange(
                    event.target.value as unknown as number,
                    "maritalStatus"
                  )
                }
                label="Marital Status"
                errors={errors.maritalStatus}
                isRequired
              >
                <MenuItem value={-1}>
                  <em>Select marital status</em>
                </MenuItem>
                {Shared.maritalStatuses.map((status) => {
                  return (
                    <MenuItem key={status.description} value={status.id}>
                      {status.description}
                    </MenuItem>
                  );
                })}
              </CustomSelect>
            </Grid>
            <Grid item xs={8}>
              <CustomSelect
                value={
                  customerProfile.drivingLicenceType?.drivingLicenceTypeId ?? -1
                }
                onChange={(event: ChangeEvent<HTMLSelectElement>) =>
                  handleCustomerProfileChange(
                    event.target.value as unknown as number,
                    "drivingLicenceType"
                  )
                }
                label="Driving License Type"
                errors={errors.drivingLicenceType}
                isRequired
              >
                <MenuItem value={-1}>
                  <em>Select type</em>
                </MenuItem>
                {Shared.licenceTypes.map((licence) => {
                  return (
                    <MenuItem key={licence.description} value={licence.id}>
                      {licence.description}
                    </MenuItem>
                  );
                })}
              </CustomSelect>
            </Grid>
            <Grid item xs={8}>
              <CustomInput
                label="Email address"
                onChange={(event: ChangeEvent<HTMLInputElement>) =>
                  handleCustomerProfileChange(
                    event.target.value,
                    "emailAddress"
                  )
                }
                value={customerProfile.emailAddress}
                errors={errors.emailAddress}
                isRequired
              />
            </Grid>
            <Grid item xs={8}>
              <CustomInput
                label="Mobile No"
                onChange={(event: ChangeEvent<HTMLInputElement>) =>
                  handleCustomerProfileChange(
                    event.target.value,
                    "mobileNumber"
                  )
                }
                value={customerProfile.mobileNumber}
              />
            </Grid>
            <Grid item xs={8}>
              <CustomInput
                label="Daytime Phone"
                onChange={(event: ChangeEvent<HTMLInputElement>) =>
                  handleCustomerProfileChange(event.target.value, "workNumber")
                }
                value={customerProfile.workNumber}
              />
            </Grid>
            <Grid item xs={8}>
              <CustomInput
                label="Evening Phone"
                onChange={(event: ChangeEvent<HTMLInputElement>) =>
                  handleCustomerProfileChange(event.target.value, "homeNumber")
                }
                value={customerProfile.homeNumber}
              />
            </Grid>
          </CardContent>
          <CardActions className="card-action">
            <Grid item xs={8}>
              <Button
                variant="contained"
                color="secondary"
                className="update-detail-btn"
                onClick={() => {
                  setHighlightRequiredDetail(true);
                  updateProfile(customerProfile);
                }}
              >
                Update Details
              </Button>
            </Grid>
          </CardActions>
        </Grid>
      </Card>
    </CustomerProfileStyles>
  );
};
