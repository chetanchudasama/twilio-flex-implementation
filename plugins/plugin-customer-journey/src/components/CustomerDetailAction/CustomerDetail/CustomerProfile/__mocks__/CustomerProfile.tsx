import React from "react";
import { CustomerDetailModel } from "@common/components";
import { CustomerProfileProps } from "../CustomerProfile";

export const CustomerProfile: React.FC<CustomerProfileProps> = (props) => (
  <div className="customer-profile">
    <button
      type="button"
      className="update-detail-btn"
      onClick={() =>
        props.updateProfile(
          Object.assign(new CustomerDetailModel(), {
            firstName: "John",
            lastName: "Fred",
            maritalStatusId: 1,
            drivingLicenceTypeId: 1,
            emailAddress: "john@gmail.com",
            dateOfBirth: new Date("06-02-1990 00:00"),
          })
        )
      }
    >
      Update Details
    </button>
  </div>
);
