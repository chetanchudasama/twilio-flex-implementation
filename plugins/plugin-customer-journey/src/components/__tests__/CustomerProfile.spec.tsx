import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import { CustomerDetailModel } from "@common/components";
import { CustomerProfile } from "../CustomerDetailAction/CustomerDetail/CustomerProfile/CustomerProfile";

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
    Shared: {
      titles: [{ id: 1, description: "Mrs" }],
      maritalStatuses: [{ id: 1, description: "Married" }],
      licenceTypes: [{ id: 1, description: "Full UK Driving Licence" }],
    },
    CustomSelect: jest.fn(({ children }) => children),
    CustomInput: jest.fn(() => {
      return <div className="custom-input" />;
    }),
    CustomerDetailModel: jest.fn().mockImplementation(() => {
      return { customerDetailModel: () => {} };
    }),
  };
});

const customerProfileModel: CustomerDetailModel = Object.assign(
  new CustomerDetailModel(),
  {
    title: { titleId: 1, titleName: "Mrs" },
    firstName: "Fred",
    lastName: "smith",
    middleName: "John",
    dateOfBirth: new Date(1990, 12, 12),
    maritalStatus: { maritalStatusId: 1, maritalStatusName: "Married" },
    drivingLicenceType: {
      drivingLicenceTypeId: 1,
      drivingLicenceTypeName: "Full UK Driving Licence",
    },
    email: "John.Fred@example.com",
    mobileNumber: "+447700903977",
    workNumber: "+444592900077",
    homeNumber: "+447492480056",
  }
);

const props = {
  customer: customerProfileModel,
  errorList: {},
  updateProfile: jest.fn(() => {}),
};

describe("CustomerProfile", () => {
  it("renders correctly", () => {
    const wrapper = create(<CustomerProfile {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should call updateProfile method, when update details button is clicked", () => {
    const wrapper = mount(<CustomerProfile {...props} />);

    wrapper.find(".update-detail-btn").first().simulate("click");
    wrapper.update();

    expect(props.updateProfile).toHaveBeenCalled();
  });
});
