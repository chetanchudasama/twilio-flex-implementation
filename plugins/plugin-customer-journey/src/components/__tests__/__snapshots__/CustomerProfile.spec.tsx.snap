// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerProfile renders correctly 1`] = `
<div
  className="css-hwenn2"
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
  >
    <div
      className="MuiGrid-container-29"
    >
      <div
        className="MuiCardContent-root-126 card-content"
      >
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65 customer-detail-title"
        >
          Customer Details
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <li
            className="MuiButtonBase-root-142 MuiListItem-root-130 MuiListItem-default-133 MuiListItem-gutters-138 MuiListItem-button-139 MuiMenuItem-root-127 MuiMenuItem-gutters-128"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="menuitem"
            tabIndex={-1}
            value={-1}
          >
            <em>
              Select title
            </em>
            <span
              className="MuiTouchRipple-root-252"
            />
          </li>
          <li
            className="MuiButtonBase-root-142 MuiListItem-root-130 MuiListItem-default-133 MuiListItem-gutters-138 MuiListItem-button-139 MuiMenuItem-root-127 MuiMenuItem-gutters-128"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="menuitem"
            tabIndex={-1}
            value={1}
          >
            Mrs
            <span
              className="MuiTouchRipple-root-252"
            />
          </li>
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <div
            className="custom-input"
          />
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <div
            className="custom-input"
          />
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <div
            className="custom-input"
          />
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65 birthDate"
        >
          <div
            className="css-1ysv0zs"
          >
            <label
              className="MuiFormLabel-root-156 MuiFormLabel-required-161 MuiInputLabel-required-149 MuiInputLabel-root-145 MuiInputLabel-animated-153 MuiInputLabel-shrink-152 label "
              data-shrink={true}
            >
              D.O.B
              <span
                className="MuiFormLabel-asterisk-162"
              >
                â€‰*
              </span>
            </label>
            <div
              className="picker"
            >
              <div
                className="MuiFormControl-root-163 MuiFormControl-fullWidth-166 date-picker"
                onClick={[Function]}
                onKeyPress={[Function]}
              >
                
                <div
                  className="MuiInputBase-root-180 MuiOutlinedInput-root-167 MuiInputBase-fullWidth-189 MuiInputBase-formControl-181 MuiInputBase-adornedEnd-185 MuiOutlinedInput-adornedEnd-171"
                  onClick={[Function]}
                >
                  <fieldset
                    aria-hidden={true}
                    className="MuiPrivateNotchedOutline-root-197 MuiOutlinedInput-notchedOutline-174"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="MuiPrivateNotchedOutline-legend-198"
                      style={
                        Object {
                          "width": 0,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input-190 MuiOutlinedInput-input-175 MuiInputBase-inputAdornedEnd-196 MuiOutlinedInput-inputAdornedEnd-179"
                    disabled={false}
                    mask={null}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    pipe={null}
                    readOnly={true}
                    required={true}
                    type="text"
                    value="12/01/1991"
                  />
                  <button
                    className="MuiButtonBase-root-265 MuiIconButton-root-259 clear-icon"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex="0"
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label-264"
                    >
                      <span
                        aria-hidden="true"
                        className="material-icons MuiIcon-root-268"
                      >
                        clear
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root-277"
                    />
                  </button>
                </div>
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <li
            className="MuiButtonBase-root-142 MuiListItem-root-130 MuiListItem-default-133 MuiListItem-gutters-138 MuiListItem-button-139 MuiMenuItem-root-127 MuiMenuItem-gutters-128"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="menuitem"
            tabIndex={-1}
            value={-1}
          >
            <em>
              Select marital status
            </em>
            <span
              className="MuiTouchRipple-root-252"
            />
          </li>
          <li
            className="MuiButtonBase-root-142 MuiListItem-root-130 MuiListItem-default-133 MuiListItem-gutters-138 MuiListItem-button-139 MuiMenuItem-root-127 MuiMenuItem-gutters-128"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="menuitem"
            tabIndex={-1}
            value={1}
          >
            Married
            <span
              className="MuiTouchRipple-root-252"
            />
          </li>
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <li
            className="MuiButtonBase-root-142 MuiListItem-root-130 MuiListItem-default-133 MuiListItem-gutters-138 MuiListItem-button-139 MuiMenuItem-root-127 MuiMenuItem-gutters-128"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="menuitem"
            tabIndex={-1}
            value={-1}
          >
            <em>
              Select type
            </em>
            <span
              className="MuiTouchRipple-root-252"
            />
          </li>
          <li
            className="MuiButtonBase-root-142 MuiListItem-root-130 MuiListItem-default-133 MuiListItem-gutters-138 MuiListItem-button-139 MuiMenuItem-root-127 MuiMenuItem-gutters-128"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="menuitem"
            tabIndex={-1}
            value={1}
          >
            Full UK Driving Licence
            <span
              className="MuiTouchRipple-root-252"
            />
          </li>
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <div
            className="custom-input"
          />
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <div
            className="custom-input"
          />
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <div
            className="custom-input"
          />
        </div>
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65"
        >
          <div
            className="custom-input"
          />
        </div>
      </div>
      <div
        className="MuiCardActions-root-223 card-action"
      >
        <div
          className="MuiGrid-item-30 MuiGrid-grid-xs-8-65 MuiCardActions-action-225"
        >
          <button
            className="MuiButtonBase-root-142 MuiButton-root-226 MuiButton-contained-237 MuiButton-containedSecondary-239 MuiButton-raised-240 MuiButton-raisedSecondary-242 update-detail-btn"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-227"
            >
              Update Details
            </span>
            <span
              className="MuiTouchRipple-root-252"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
